stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -388,6 +388,126 @@ const ⋅ = dot
 const × = cross
 export ⋅, ×
 
+"""
+    hadamard(a, b)
+    a ⊙ b
+
+Elementwise multiplication of `a` and `b`. This yields the same result as `map(*, a, b)`
+if `a` and `b` have the same axes, and throws an error otherwise.
+
+`⊙` can be passed as an operator to higher-order functions.
+
+```jldoctest
+julia> a = [2, 3]; b = [5, 7];
+
+julia> a ⊙ b
+2-element Array{$Int,1}:
+ 10
+ 21
+
+julia> a ⊙ [5]
+ERROR: DimensionMismatch("in the axes of `A` and `B` must match, got (Base.OneTo(2),) and (Base.OneTo(1),)")
+[...]
+```
+
+!!! compat "Julia 1.5"
+    This function requires at least Julia 1.5. In Julia 1.0-1.4 it is available from
+    the `Compat` package.
+"""
+function hadamard(A, B)
```

Comment:
Should this be restricted to `AbstractArray` only? A method error might be slightly clearer than erroring because `axes(A)` is not defined.

Reply:
Seems like a good idea. Incorporating https://github.com/JuliaLang/julia/pull/35150#issuecomment-619522055, that would open the door to
```julia
hadamard(a, b) = a * b
```
which would allow it to handle `hadamard(A::ArrayOfNumbers, B::ArrayOfArrays)`.

Reply:
The question is still whether we actually want that, or if we want to have the restriction that `hadamard` only ever does elementwise multiplication and doesn't extend dimensions. That would also mean that the two arrays would have to be nested equally deeply. Though I could definitely see `hadamard(A::ArrayOfNumbers, B::ArrayOfArrays)` being useful.

Reply:
> only ever does elementwise multiplication and doesn't extend dimensions

That's all this allows:
```julia
julia> a = [1, 2, 3]
3-element Array{Int64,1}:
 1
 2
 3

julia> b = [[1,2], [3,4], [5,6]]
3-element Array{Array{Int64,1},1}:
 [1, 2]
 [3, 4]
 [5, 6]

julia> a ⊙ b
3-element Array{Array{Int64,1},1}:
 [1, 2]
 [6, 8]
 [15, 18]

julia> reshape(a, 3, 1) ⊙ b
ERROR: DimensionMismatch("Axes of `A` and `B` must match, got (Base.OneTo(3), Base.OneTo(1)) and (Base.OneTo(3),)")
Stacktrace:
 [1] (::LinearAlgebra.var"#throw_dmm#198")(::Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}, ::Tuple{Base.OneTo{Int64}}) at /home/tim/src/julia-master/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/LinearAlgebra.jl:418
 [2] hadamard(::Array{Int64,2}, ::Array{Array{Int64,1},1}) at /home/tim/src/julia-master/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/LinearAlgebra.jl:421
 [3] top-level scope at REPL[8]:1
```

Reply:
Currently yes, but I was talking talking about your proposal of allowing `hadamard(a, b) = a * b`.

Reply:
That's already in this PR, and the demo above requires it.

Most likely, I am failing to understanding your concern. Can you give a concrete example where the current implementation in this PR is problematic?